adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsApplicationCodeGenerator.java

```diff
@@ -34,67 +34,91 @@
-                .param(DEFAULT_ANNOTATION_PARAMETER, defaultIfBlank(baseUriPathWithoutContext(raml), "/"));
+    /**
+     * Generate the implementation of {@link Application} and set the {@link ApplicationPath}.
+     *
+     * @param raml the RAML document being generated from
+     * @return the {@link TypeSpec.Builder} that defines the class
+     */
+    private TypeSpec.Builder classSpecFor(final Raml raml) {
+        return classBuilder(applicationNameOf(raml))
+                .addModifiers(PUBLIC)
+                .superclass(Application.class)
+                .addAnnotation(AnnotationSpec.builder(ApplicationPath.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$S", defaultIfBlank(baseUriPathWithoutContext(raml), "/"))
+                        .build());
     }
 
-    private void addSuperClass(final JDefinedClass application) {
-        application._extends(Application.class);
+    /**
+     * Generate the getClasses method that returns the set of implemented resource classes.
+     *
+     * @param implementationNames the collection of implementation class names
+     * @return the {@link MethodSpec} that represents the getClasses method
+     */
+    private MethodSpec generateGetClassesMethod(final Collection<String> implementationNames) {
+        final ParameterizedTypeName wildcardClassType = ParameterizedTypeName.get(ClassName.get(Class.class), WildcardTypeName.subtypeOf(Object.class));
+        final ParameterizedTypeName classSetType = ParameterizedTypeName.get(ClassName.get(Set.class), wildcardClassType);
+        final ParameterizedTypeName classHashSetType = ParameterizedTypeName.get(ClassName.get(HashSet.class), wildcardClassType);
+
+        return MethodSpec.methodBuilder("getClasses")
+                .addModifiers(PUBLIC)
+                .addAnnotation(Override.class)
+                .addCode(CodeBlock.builder()
+                        .addStatement("$T classes = new $T()", classSetType, classHashSetType)
+                        .add(codeStatementForEach(implementationNames))
+                        .addStatement("return classes")
+                        .build())
+                .returns(classSetType)
+                .build();
     }
 
-    private void addMethods(final JDefinedClass application, final Collection<String> implementationNames) {
-        JType wildcardClassType = codeModel.ref(Class.class).narrow(codeModel.wildcard());
-        JType classSetType = codeModel.ref(Set.class).narrow(wildcardClassType);
-
-        JMethod getClassesMethod = application.method(PUBLIC, classSetType, "getClasses");
+    /**
+     * Generate code to add each resource implementation class to the classes hash set.
+     *
+     * @param implementationNames the collection of implementation class names
+     * @return the {@link CodeBlock} that represents the generated statements
+     */
+    private CodeBlock codeStatementForEach(final Collection<String> implementationNames) {
```

Comment:
More descriptive method name? - What is the code statement being added (not clear from calling method)